### PR TITLE
Support sentinel username in `SentinelConfiguration`

### PR DESCRIPTION
**Pull request by vikasgarghb**
- [ X ] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [ X ] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ X ] You submit test cases (unit or integration tests) that back your changes.
- [ X ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

This pull request allows you to provide username for sentinel which in turn allows support for ACL based authentication provided with redis 6.

This covers the issue mentioned here: https://github.com/spring-projects/spring-data-redis/issues/2218

### CODE REVIEW

File: src/main/java/org/springframework/data/redis/connection/lettuce/LettuceConverters.java [hunk 1]
```diff
@@ -512,7 +513,13 @@ public static RedisURI sentinelConfigurationToRedisURI(RedisSentinelConfiguratio
 
 			RedisURI.Builder sentinelBuilder = RedisURI.Builder.redis(sentinel.getHost(), sentinel.getPort());
 
-			sentinelPassword.toOptional().ifPresent(sentinelBuilder::withPassword);
+			String sentinelUsername = sentinelConfiguration.getSentinelUsername();
+			if (StringUtils.hasText(sentinelUsername)) {
```
Comment (mp911de) [inline src/main/java/org/springframework/data/redis/connection/lettuce/LettuceConverters.java:517]: We should set the authentication with username only if a password is present in addition to the username check. Having a username without a password isn't supported in the authentication scheme.

File: src/test/java/org/springframework/data/redis/connection/lettuce/LettuceConvertersUnitTests.java [hunk 2]
```diff
@@ -252,4 +255,18 @@ void convertsExpirationToGetExPXAT() {
...
+		RedisSentinelConfiguration sentinelConfiguration = new RedisSentinelConfiguration()
+				.master(MASTER_NAME)
+				.sentinel("127.0.0.1", 26379)
+				.sentinel("127.0.0.1", 26380);
+		sentinelConfiguration.setSentinelUsername("admin");
+		sentinelConfiguration.setSentinelPassword(RedisPassword.of("88888888-8x8-getting-creative-now"));
+		sentinelConfiguration.setUsername("app");
+		sentinelConfiguration.setPassword(RedisPassword.of("88888888-8x8-getting-creative-now"));
+		RedisURI redisURI = LettuceConverters.sentinelConfigurationToRedisURI(sentinelConfiguration);
+		assertThat(redisURI.toString()).isEqualTo("redis-sentinel://app:*********************************@127.0.0.1,127.0.0.1:26380?sentinelMasterId=mymaster");
```
Comment (mp911de) [inline src/test/java/org/springframework/data/redis/connection/lettuce/LettuceConvertersUnitTests.java:270]: Please rewrite this check to inspect the `password` property of each sentinel URI (`RedisURI.getSentinels()`). The `toString` check performs a very strict test on unrelated internals and also the rendering of the password isn't part of any API contract.

**Review by mp911de — CHANGES_REQUESTED** [review]
Thanks for your pull request. I added comments regarding minor issues. Also, the formatting seems odd, care to have a look? You can find the formatter settings at https://github.com/spring-projects/spring-data-build/tree/main/etc/ide

**Comment by vikasgarghb** [comment]
@mp911de Can this be approved, merged and made part of the next release?

**Comment by mp911de** [comment]
Thanks a lot. I'm picking up this PR from here.

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged and polished now.

**Pull request closed without merging**